### PR TITLE
[WEJBHTTP-144] Add system property org.wildfly.httpclient.tcp-nodelay for ennabling/disabling Options.TCP_NODELAY

### PR DESCRIPTION
**Pull request by rachmatowicz**
PR for creating patch of wildfly-http-client 1.1.17.Final for EAP 7.4.18.GA which modifies option map.

### CODE REVIEW

File: common/src/main/java/org/wildfly/httpclient/common/WildflyHttpContext.java [hunk 1]
```diff
@@ -44,6 +45,7 @@
 public class WildflyHttpContext implements Contextual<WildflyHttpContext> {
 
     private static final int LEAK_DETECTION = Integer.getInteger("org.wildfly.http-client.buffer-leak-detection", 0);
+    private static final boolean TCP_NODELAY = Boolean.getBoolean("org.wildfly.httpclient.tcp-nodelay");
```
Comment (jbaesner) [inline common/src/main/java/org/wildfly/httpclient/common/WildflyHttpContext.java:48]: Please change the system property name to match the already existing naming scheme to `org.wildfly.http-client.tcp-nodelay`

Comment (ropalka) [reply common/src/main/java/org/wildfly/httpclient/common/WildflyHttpContext.java:48]: Please change from "org.wildfly.httpclient.tcp-nodelay" to "org.wildfly.httpc-lient.tcp-nodelay" @rachmatowicz

Comment (fl4via) [reply common/src/main/java/org/wildfly/httpclient/common/WildflyHttpContext.java:48]: Yes, we need to make it consistent with the other properties in the project.

**Review by fl4via — CHANGES_REQUESTED** [review]

**Comment by fl4via** [comment]
Yes @aogburn . Please create a follow up Jira and we can follow suit.

**Comment by fl4via** [comment]
@rachmatowicz because this fix is needed for an urgent release, I'm taking the freedom of editing it and moving it to ready for review. Thanks for the PR!

**Review by fl4via — APPROVED** [review]

**Pull request merged**